minimal-plugin/src/main/java/io/micronaut/gradle/MicronautRuntimeDependencies.java

```diff
@@ -0,0 +1,136 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.gradle;
+
+import static io.micronaut.gradle.MicronautRuntime.*;
+
+/**
+ * Resolves the dependencies for the current runtime and application type
+ * @author Sergio del Amo
+ * @since 3.4.0
+ */
+public final class MicronautRuntimeDependencies {
+
+    private static final String GROUP_MICRONAUT = "io.micronaut";
+    private static final String ARTIFACT_ID_MICRONAUT_SERVER_NETTY = "micronaut-http-server-netty";
+    private static final String GROUP_MICRONAUT_SERVLET = "io.micronaut.servlet";
+    private static final String GROUP_MICRONAUT_AWS = "io.micronaut.aws";
+    private static final String GROUP_MICRONAUT_GCP = "io.micronaut.gcp";
+    private static final String GROUP_GOOGLE_CLOUD_FUNCTIONS = "com.google.cloud.functions";
+    private static final String ARTIFACT_ID_FUNCTIONS_FRAMEWORK_API = "functions-framework-api";
+
+    private static final String GROUP_MICRONAUT_ORACLE = "io.micronaut.oraclecloud";
+    private static final String ARTIFACT_ID_MICRONAUT_ORACLE_HTTP = "micronaut-oraclecloud-function-http";
+    private static final String ARTIFACT_ID_MICRONAUT_ORACLE_HTTP_TEST = "micronaut-oraclecloud-function-http-test";
+
+    private static final String ARTIFACT_ID_MICRONAUT_GCP_FUNCTION_HTTP = "micronaut-gcp-function-http";
+    private static final String ARTIFACT_ID_MICRONAUT_GCP_FUNCTION_HTTP_TEST = "micronaut-gcp-function-http-test";
+
+    private static final String GROUP_MICRONAUT_AZURE = "io.micronaut.azure";
+    private static final String ARTIFACT_ID_MICRONAUT_AZURE_FUNCTION_HTTP = "micronaut-azure-function-http";
+    private static final String ARTIFACT_ID_MICRONAUT_AZURE_FUNCTION_HTTP_TEST = "micronaut-azure-function-http-test";
+
+    private static final String ARTIFACT_ID_MICRONAUT_AWS_CUSTOM_RUNTIME = "micronaut-function-aws-custom-runtime";
+    private static final String ARTIFACT_ID_MICRONAUT_AWS_API_PROXY = "micronaut-function-aws-api-proxy";
+    private static final String ARTIFACT_ID_MICRONAUT_AWS_API_PROXY_TEST = "micronaut-function-aws-api-proxy-test";
+
+    private static final String ARTIFACT_ID_MICRONAUT_SERVLET_JETTY = "micronaut-http-server-jetty";
+    private static final String ARTIFACT_ID_MICRONAUT_SERVLET_TOMCAT = "micronaut-http-server-tomcat";
+    private static final String ARTIFACT_ID_MICRONAUT_SERVLET_UNDERTOW = "micronaut-http-server-undertow";
+    private static final String COLON = ":";
+
+    private MicronautRuntimeDependencies() {
+    }
+
+    /**
+     * @param runtime Micronaut runtime
+     * @return The dependencies and scopes
+     */
+    public static Dependencies findApplicationPluginDependenciesByRuntime(MicronautRuntime runtime) {
+        if (runtime == LAMBDA_JAVA || isLambdaProvided(runtime)) {
```

Comment:
:o: Maybe add a `isLambda` method to `MicronautRuntime`.

Reply:
do you want me to replace `runtime == LAMBDA_JAVA` with `runtime.isLambdaJava()`?

_MicronautRuntime_
```

    public boolean isLambdaJava() {
        return this == LAMBDA_JAVA;
    }
```

Please, note that I have added an `isLambdaProvided()` to encapsulate the deprecation of `LAMBDA` runtime .

Reply:
No, I want a method which says "the runtime is a lambda variant" which encapsulates all possible lambda cases.

Reply:
Resolved via https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/434/commits/62a7d53a58e03383ae79e07c725c68e49d48050d